learn/src/main/java/org/navgurukul/learn/ui/common/Util.kt

```diff
@@ -0,0 +1,15 @@
+package org.navgurukul.learn.ui.common
+
+import android.content.Context
+import android.net.ConnectivityManager
+import android.net.NetworkInfo
+
+@Suppress("DEPRECATION")
```

Comment:
Should not use any deprecated APIs. We should handle it properly with permission and stuff which Android recommends.

Reply:
Else code block is still deprecated. It require a network callback listener which need to be registered at application level. This can be done at app level and all different module can use that.
```
fun isOnline(context: Context): Boolean {
        val connectivityManager =
            context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
            val capabilities =
                connectivityManager.getNetworkCapabilities(connectivityManager.activeNetwork)
            if (null != capabilities)
                return capabilities.hasCapability(NET_CAPABILITY_INTERNET)
        } else {
            val cm = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
            val activeNetwork: NetworkInfo? = cm.activeNetworkInfo
            return activeNetwork?.isConnectedOrConnecting!!
        }
        return false
    }
```